### PR TITLE
[ACTIONS/SOURCES] Etsy new actions sources

### PR DESCRIPTION
**Pull request by jcortes**
### <samp>🤖 Generated by Copilot at f6f0efc</samp>

This pull request adds new action and source components for the Etsy channel, as well as common modules, constants, and utilities for the Etsy app. The action components allow users to create, delete, get, and update listings and their properties. The source components emit events when listings, shop receipts, or transactions are updated or created. The common modules define the app props, methods, and authentication, as well as the base and polling props and methods for the sources. The constants and utilities store the common values and functions used by the Etsy components. The pull request also updates the `package.json` file and deletes the unused `.gitignore` file.

### <samp>🤖 Generated by Copilot at f6f0efc</samp>

*  Add new action components to create, delete, get, and update listings and listing properties on Etsy ([link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-41ad4988f7f97432b10ad38a7bd7bde109dfe329fe00e85b86d2ca25b9f76addR1-R107), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-a7e6bd14f89f760d5f70573571de30bffd1c72679e9e13feeed05e20ba75e63fR1-R38), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-e97234d08ff04427cd80c3e83fe8e45911a0201559286b8ffba3b5690661f160R1-R38), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-8e9e1c52eda9ab2114070c29d644ea17554e352bd84489cdae951281c2ca437aR1-R69))
* Add new source components to emit events when listings, shop receipts, and transactions are updated or created on Etsy ([link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-3d3d671d12ae301c6d0455e5b8f6c1dccae1eedb1c822ca376262b5701a46738R1-R35), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-3ef54f07e2b46de22bcac03324b53f8f3515c183545a5ac3770588debf03b305R1-R33), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-0186f3a3d1d2ff0d8b4e10f4f7b06ffb59dfbf6a601369d0e0a24ab985ecf82aR1-R33))
* Add new common modules to store constants, utility functions, and base classes for the Etsy components ([link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-5041e98ead22fb006a9999ca2d9807ecb703ef0582e5a409ee68bb1993cefd2cR1-R80), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-75fbcca27dd3c4f19034171443763c4bf33dde4c201d3d1a4ebcf2c6de513edbR1-R12), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-f348a2023866430c14c8412cb12f2c6d675de42aaaf73484d91aecff66cf068eR1-R14), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-feb30b8042e08c0b28aec23e0ccdcb461104dc635ce3b3b43a4bb2fd8968e259R1-R72))
* Add new app module to define the app type, app name, prop definitions, and methods for the Etsy components ([link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-464821e1b0012d13c6099646e3cf206d190923970f1f7334879b22fcd7da0243R1-R266))
* Update the `package.json` file of the Etsy components to change the main entry point, add the license and dependencies fields, and bump the version to 0.1.0 ([link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-329e33e0b59541b44dc1bd8413c6b00f7a5ab7ab635a6113c134311829668131L3-R19))
* Delete unused files ([link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-0db9c02d265577dd32fc3aa3b5b18a0f03426657c9f413e3ac1acf29e978c8bb), [link](https://github.com/PipedreamHQ/pipedream/pull/7312/files?diff=unified&w=0#diff-547ca69cedb5bee5bd3f3c9e4c298727bc2bd798444e61052772c6bc69d43a99))

### CODE REVIEW

File: components/etsy/sources/common/polling.mjs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    async processStreamEvents(resourcesStream) {
+      const resources = await utils.streamIterator(resourcesStream);
+
+      // const [
+      //   lastResource,
+      // ] = resources;
+
+      // if (lastResource?.created_timestamp) {
+      //   this.setLastCreatedAt(lastResource.created_timestamp);
+      // }
```
Comment (michelle0927) [inline components/etsy/sources/common/polling.mjs:51]:
```suggestion
```

File: components/etsy/package.json [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-  "main": "dist/app/etsy.app.mjs",
+  "main": "etsy.app.mjs",
   "keywords": [
     "pipedream",
     "etsy"
   ],
-  "files": ["dist"],
   "homepage": "https://pipedream.com/apps/etsy",
   "author": "Pipedream <support@pipedream.com> (https://pipedream.com/)",
+  "license": "MIT",
```
Comment (michelle0927) [inline components/etsy/package.json:12]:
```suggestion
```

**Review by michelle0927 — COMMENTED** [review]
Looks good! Just a couple minor comments. Moving to QA.

**Review by vellames — DISMISSED** [review]
lgtm

**Review by vellames — APPROVED** [review]

**Pull request merged**